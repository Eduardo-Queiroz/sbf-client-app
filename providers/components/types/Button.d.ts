interface ButtonPropsInterface {
  onPress: () => void;
  text: string;
  disabled?: boolean;
}

declare type ButtonProps = ButtonPropsInterface & TrackeablePropsInterface;
