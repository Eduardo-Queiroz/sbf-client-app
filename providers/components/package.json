{
  "name": "@sbf-providers/components",
  "version": "1.0.0",
  "author": "Eduardo Queiroz",
  "license": "ISC",
  "main": "./src/index.tsx"
}
