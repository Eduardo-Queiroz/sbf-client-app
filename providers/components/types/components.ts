export enum ComponentsType {
  LOADING = 'LOADING',
  BUTTON = 'BUTTON',
  FOOTER = 'FOOTER',
}
