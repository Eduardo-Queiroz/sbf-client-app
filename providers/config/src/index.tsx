import React, {PureComponent} from 'react';

export type Config = {[key in string]: string};

interface ConfigContext {
  config?: Config;
}

export const context = React.createContext<ConfigContext & LayoutProps>({});

export class ConfigProvider extends PureComponent<ConfigContext> {
  public render() {
    const {config} = this.props;
    return (
      <context.Provider value={{config}}>
        {this.props.children}
      </context.Provider>
    );
  }
}

export const useConfig = () => {
  const {config} = React.useContext<ConfigContext>(context);
  if (!config) throw new Error(`Provider de configuração não inicializado`);
  return config;
};
