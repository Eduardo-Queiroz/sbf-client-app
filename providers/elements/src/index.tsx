import React, {PureComponent} from 'react';
import {ElementsType} from '../types/elements';
export {ElementsType} from '../types/elements';

export type Element = {[key in ElementsType]: React.FC<any>};

interface ElementsContext {
  elements?: Element;
}

export const context = React.createContext<ElementsContext>({});

export class ElementProvider extends PureComponent<ElementsContext> {
  public render() {
    const {elements, children} = this.props;
    return (
      <context.Provider
        value={{
          elements,
        }}>
        {children}
      </context.Provider>
    );
  }
}
export const useElement = <PropsType extends unknown>(
  type: ElementsType,
): React.FC<PropsType> => {
  const {elements} = React.useContext<ElementsContext>(context);
  if (!elements) throw new Error(`Providers de elementos n√£o inicializados`);
  return elements[type] || <></>;
};
