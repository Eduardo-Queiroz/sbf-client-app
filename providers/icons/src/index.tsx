import React, {PureComponent} from 'react';
import {IconsType} from '../types';
export {IconsType} from '../types';

export type Icon = {[key in IconsType]: React.FC<any>};

interface IconContext {
  icons?: Icon;
}

export const context = React.createContext<IconContext>({});

export class IconProvider extends PureComponent<IconContext> {
  public render() {
    const {icons} = this.props;
    return (
      <context.Provider value={{icons}}>{this.props.children}</context.Provider>
    );
  }
}

export const useIcon = (type: IconsType) => {
  const {icons} = React.useContext<IconContext>(context);
  if (!icons) throw new Error(`Provider de icones n√£o inicializados`);
  return icons[type] || null;
};
