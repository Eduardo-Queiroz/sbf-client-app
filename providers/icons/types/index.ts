export enum IconsType {
  ICON_MINUS = 'ICON_MINUS',
  ICON_PLUS = 'ICON_PLUS',
  ICON_CLOSE = 'ICON_CLOSE',
  ICON_CART = 'ICON_CART',
  ICON_NO_CART = 'ICON_NO_CART',
  ICON_SEARCH = 'ICON_SEARCH',
  ICON_STAR = 'ICON_STAR',
  ICON_SAD_WARNING = 'ICON_SAD_WARNING',
  ICON_HAPPY_WARNING = 'ICON_HAPPY_WARNING',
}
