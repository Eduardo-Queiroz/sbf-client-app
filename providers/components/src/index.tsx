import React, {PureComponent} from 'react';
import {ComponentsType} from '../types/components';
export {ComponentsType} from '../types/components';

type Component = {[key in ComponentsType]: React.FC<any>};

interface ComponentsContext {
  components?: Component;
}

export const context = React.createContext<ComponentsContext>({});

export class ComponentProvider extends PureComponent<ComponentsContext> {
  public render() {
    const {components, children} = this.props;
    return (
      <context.Provider
        value={{
          components,
        }}>
        {children}
      </context.Provider>
    );
  }
}

export const useComponent = <PropsType extends unknown>(
  type: ComponentsType,
): React.FC<PropsType> => {
  const {components} = React.useContext<ComponentsContext>(context);
  if (!components)
    throw new Error(`Providers de componentes n√£o inicializados`);
  return components[type] || <></>;
};
