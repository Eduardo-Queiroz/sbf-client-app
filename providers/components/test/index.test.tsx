/* eslint-disable react/display-name */
import React from 'react';
import {ComponentProvider, useComponent, context} from './../src/index';
import {ComponentsType} from './../types/components';
import {render} from '@testing-library/react-native';
import {renderHook} from '@testing-library/react-hooks';

const valueWrapper = {
  [ComponentsType.LOADING]: true,
  [ComponentsType.BUTTON]: true,
};

const makeWrapper = ({children}: {children?: any}) => (
  <context.Provider
    value={{
      components: valueWrapper,
    }}>
    {children}
  </context.Provider>
);

describe('it should test components provider', () => {
  test('it should verify provider snapshot', () => {
    const {toJSON} = render(<ComponentProvider components={valueWrapper} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('it should verify empty provider context', () => {
    try {
      const {result} = renderHook(() => useComponent(ComponentsType.BUTTON));
      expect(result).toBe(false);
    } catch (e) {
      expect(e.message).toBe('Providers de componentes nÃ£o inicializados');
    }
  });

  test('it should verify specific value from provider context', async () => {
    const {result} = renderHook(() => useComponent(ComponentsType.BUTTON), {
      wrapper: makeWrapper,
    });
    expect(result.current).toBe(true);
  });
});
