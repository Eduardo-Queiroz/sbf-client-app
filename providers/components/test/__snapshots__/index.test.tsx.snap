// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should test components provider it should verify provider snapshot 1`] = `null`;
